pkg/k8shandler/service_monitor.go

```diff
@@ -0,0 +1,76 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/client/monitoring/v1"
+	v1alpha1 "github.com/openshift/elasticsearch-operator/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateOrUpdateServiceMonitors ensures the existence of ServiceMonitors for Elasticsearch cluster
+func CreateOrUpdateServiceMonitors(dpl *v1alpha1.Elasticsearch) error {
+	serviceMonitorName := fmt.Sprintf("monitor-%s-%s", dpl.Name, "cluster")
+	owner := asOwner(dpl)
+
+	labelsWithDefault := appendDefaultLabel(dpl.Name, dpl.Labels)
+
+	err := createOrUpdateServiceMonitor(serviceMonitorName, dpl.Namespace, labelsWithDefault, owner)
+	if err != nil {
+		return fmt.Errorf("Failure creating service %v", err)
+	}
+
+	return nil
+}
+
+func createOrUpdateServiceMonitor(serviceMonitorName, namespace string, labels map[string]string, owner metav1.OwnerReference) error {
```

Comment:
can we move this logic up into the public scoped function since it only does a single operation for creating the service monitors